### ISSUE TITLE
Remove fork of CompactValue.h from ReactCommon

### ISSUE THREAD

**Issue by htpiv** [microsoft/react-native-windows#10145]
### Problem Description

CompactValue.h is being forked to avoid undefined behavior when compiling with VC++ when using `/fp:precise` (we previously worked around the bad codegen by using `/fp:strict`, but in addition to that generating slower code, it is also causing layout issues and crashes in Office when the floating point control register is in an unexpected state)

### Steps To Reproduce

* Look at ReactCommon\TEMP_UntilReactCommonUpdate\yoga\yoga\CompactValue.h
* 😥😥😥

### CLI version

8.0.2

### Environment

```markdown
info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 Intel(R) Xeon(R) CPU E5-1680 v4 @ 3.40GHz
    Memory: 37.33 GB / 63.90 GB
  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowAllTrustedApps: Enabled
      AllowDevelopmentWithoutDevLicense: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 17.2.32602.215 (Visual Studio Enterprise 2022)
  Languages:
    Java: 1.8.0_302 - C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by rozele** [comment]
Can be resolved once RNW pulls in fbf7a6feb53c6d2e3abae4bdb54b2390c229f2cf

**Comment by TatianaKapos** [comment]
Was removed in #10479
